pkg/reconciler/integrationsource/integrationsource.go

```diff
@@ -0,0 +1,85 @@
+package integrationsource
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "knative.dev/eventing/pkg/apis/sources/v1"
+	"knative.dev/eventing/pkg/apis/sources/v1alpha1"
+	clientset "knative.dev/eventing/pkg/client/clientset/versioned"
+	"knative.dev/eventing/pkg/client/injection/reconciler/sources/v1alpha1/integrationsource"
+	v1listers "knative.dev/eventing/pkg/client/listers/sources/v1"
+	listers "knative.dev/eventing/pkg/client/listers/sources/v1alpha1"
+	"knative.dev/eventing/pkg/reconciler/integrationsource/resources"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+)
+
+const (
+	// Name of the corev1.Events emitted from the reconciliation process
+	containerSourceCreated = "ContainerSourceCreated"
+	containerSourceUpdated = "ContainerSourceUpdated"
+)
+
+// Reconciler implements controller.Reconciler for ContainerSource resources.
+type Reconciler struct {
+	kubeClientSet     kubernetes.Interface
+	eventingClientSet clientset.Interface
+
+	containerSourceLister   v1listers.ContainerSourceLister
+	integrationSourceLister listers.IntegrationSourceLister
+}
+
+// Check that our Reconciler implements Interface
+var _ integrationsource.Interface = (*Reconciler)(nil)
+
+func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1alpha1.IntegrationSource) pkgreconciler.Event {
```

Comment:
will we add reconciler unit tests for this one?

Reply:
sure